exercises/src/main/scala/fpinscala/state/State.scala

```diff
@@ -30,34 +32,124 @@ object RNG {
       (f(a), rng2)
     }
 
-  def nonNegativeInt(rng: RNG): (Int, RNG) = ???
+  // 6.1
+  // Use RNG.nextInt to generate a random number between 0 and Int.MaxValue inclusive.  Corner case
+  // when nextInt returns Int.MinValue, which doesn't have a negative counterpart
+  def nonNegativeInt(rng: RNG): (Int, RNG) = {
+    val (i, r) = rng.nextInt
+    if (i < 0) (-(i + 1), r) else (i, r)
```

Comment:
This is pretty much the official answer they have in the git repo, so that's good enough. But I would argue that a better answer would be to exclude `Int.MinValue` and get the next value instead, because otherwise, the value 0 is twice as likely as any other values to be randomly generated. I suppose it's a minor issue though, and one that isn't really related to what this exercise is trying to demonstrate anyway.